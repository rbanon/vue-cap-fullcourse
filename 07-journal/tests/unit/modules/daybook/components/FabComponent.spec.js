import { shallowMount } from "@vue/test-utils";
import FabComponent from "@/modules/daybook/components/FabComponent.vue";

describe("Pruebas en el FabComponent", () => {
  const wrapper = shallowMount(FabComponent);
  it("debe de renderizar el componente correctamente", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Debe de mostrar el ícono por defecto", () => {
    const iTag = wrapper.find("i");
    expect(iTag.classes("fa-plus")).toBeTruthy();
  });

  it("Debe de mostrar el ícono por argumento: fa-circle", () => {
    const wrapper = shallowMount(FabComponent, {
      props: {
        icon: "fa-circle",
      },
    });
    const iTag = wrapper.find("i");
    expect(iTag.classes("fa-circle")).toBeTruthy();
  });

  it("Debe de emitir el evento on:click cuando se hace click", () => {
    wrapper.find("button").trigger("click");
    expect(wrapper.emitted("on:click")).toHaveLength(1);
  });
});
