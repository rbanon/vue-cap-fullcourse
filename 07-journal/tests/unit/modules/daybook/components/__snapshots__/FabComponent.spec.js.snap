// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el FabComponent debe de renderizar el componente correctamente 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
